crates/rethnet_rpc_server/src/hardhat_methods.rs

```diff
@@ -0,0 +1,766 @@
+    #[serde(
+        rename = "hardhat_setCoinbase",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetCoinbase(Address),
+    /// hardhat_setLoggingEnabled
+    #[serde(
+        rename = "hardhat_setLoggingEnabled",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetLoggingEnabled(bool),
+    /// hardhat_setMinGasPrice
+    #[serde(
+        rename = "hardhat_setMinGasPrice",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetMinGasPrice(U256),
+    /// hardhat_setNextBlockBaseFeePerGas
+    #[serde(
+        rename = "hardhat_setNextBlockBaseFeePerGas",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetNextBlockBaseFeePerGas(U256),
+    /// hardhat_setNonce
+    #[serde(rename = "hardhat_setNonce")]
+    SetNonce(Address, U256),
+    /// hardhat_setPrevRandao
+    #[serde(
+        rename = "hardhat_setPrevRandao",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetPrevRandao(ZeroXPrefixedBytes),
+    /// hardhat_setStorageAt
+    #[serde(rename = "hardhat_setStorageAt")]
+    SetStorageAt(Address, U256, U256),
+    /// hardhat_stopImpersonatingAccount
+    #[serde(
+        rename = "hardhat_stopImpersonatingAccount",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    StopImpersonatingAccount(Address),
+}
+
+#[cfg(test)]
+mod tests {
+    use bytes::Bytes;
+
+    use super::*;
+
+    fn help_test_method_invocation_serde(call: HardhatMethodInvocation) {
+        let json = serde_json::json!(call).to_string();
+
+        #[derive(Debug, serde::Deserialize)]
+        struct MethodInvocationStructWithUntypedParams {
+            #[allow(dead_code)]
+            method: String,
+            #[allow(dead_code)]
+            params: Vec<serde_json::Value>,
+        }
+        #[derive(Debug, serde::Deserialize)]
+        #[serde(untagged)]
+        enum MethodInvocationEnumWithUntypedParams {
+            Eth(MethodInvocationStructWithUntypedParams),
+            Hardhat(MethodInvocationStructWithUntypedParams),
+        }
+        serde_json::from_str::<MethodInvocationEnumWithUntypedParams>(&json).unwrap_or_else(|_| {
+            panic!(
+                "should have successfully deserialized, with params as a Vec<String>, json {json}"
+            )
+        });
+
+        let call_decoded: HardhatMethodInvocation = serde_json::from_str(&json)
+            .unwrap_or_else(|_| panic!("should have successfully deserialized json {json}"));
+        assert_eq!(call, call_decoded);
+    }
+
+    #[test]
+    fn test_serde_hardhat_add_compilation_result() {
+        // these were taken from a run of TypeScript function compileLiteral
+        let compiler_input_json = r#"{
```

Comment:
I'm curious what your thought process was in including the input as a literal here.

One possible alternative would be to use move this to a fixture and use https://doc.rust-lang.org/std/macro.include_str.html to include it instead?

Curious to hear your thoughts

Reply:
I didn't put much thought into it at all, to be honest. It took some effort to get it this far, so I was just going for the most expedient path. Honestly I didn't even notice that the `compiler_output_json` was so huge, because my editor was folding it for me. I'm up for moving it to a fixture if you think that's the right thing to do.

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/3998/commits/cf97805377e5fcc749a8bf71905d04073a7bc6be